app/core/auth/__init__.py

```diff
@@ -60,14 +84,15 @@ async def wrapper(*args, **kwargs):
             ).eq(
             'user_type', user_data.user.user_metadata.get('user_type')
             ).limit(1).single().execute()
-        label = result.data.get('sources')
+        labels = result.data.get('sources')
 
-        if not label:
+        if not labels:
             raise PermissionException(
-            "Unauthorized access. User does not have access to any sources."
+            "Unauthorized access. User does not have access to any sources." 
+            # Or we could just give open access sources?
             )
```

Comment:
Yes, if we plan to allow unregistered users and let them access the open-access sources, I guess we need to avoid raising errors here and in the other decorator where user token is checked. The code above may also have to be changed to expect None as user token.

Reply:
OK yep, I've just done that now.